content/tutorials/channel-enumeration-rest.textile

```diff
@@ -0,0 +1,174 @@
+    Ably Channel Enumeration using REST API - Demo
+    <br/>
+    <div>
+        <button id='enumerate' onclick="enumerateChannels()">Enumerate channels</button>
+        <br>
+        <textarea id="result" rows="30" style="width: 30%; margin-top: 14px; font-family: courier, courier new; background-color: #333; color: orange;  overflow-y: scroll;"
+            disabled>
+        </textarea>
+    </div>
+    <script src="main.js"></script>
+</body>
+
+</html>
+```
+
+
+The key thing to note in the HTML above is the inclusion of two JS files, one is the Ably Library, referenced via the CDN, while the other is the @main.js@ file which will include our logic. We'll work on this next.
+
+
+h2(#performing-the-rest-request). Step 4 - Using Ably's REST library to perform a REST API request
+
+As mentioned before, channel enumeration can currently be implemented only via our "REST API":/rest-api. However, the "REST library":/rest offers a "request() method":https://www.ably.io/documentation/rest/usage#request to perform REST requests directly with our raw REST API.
+
+For the simplicity of this tutorial, we'll use "Basic authentication":/core-features/authentication#basic-authentication to perform the request. However, it is highly recommended to use "Token auth":/core-features/authentication#token-authentication on client-side applications for "better security and protection of your API key":/root/best-practice-guide#using-api-key-client-side.
+
+In order to receive channel enumeration information, we'll have to make use of the @/channels@ endpoint of "Ably's REST API":rest-api. We'll perform a @GET@ request using the REST library, and the result of this request will be an "HTTPPaginatedResponse":https://www.ably.io/documentation/rest/usage#http-paginated-response object. This allows the results to be returned in a paginated fashion, i.e, it returns a page of results and allows you to automatically request for more pages until all of the available results are returned.
+
+Let's begin with instantiating the Ably REST library using the API key. Create a new file called @main.js@ and add the following to it.
+
+```[javascript]
+var ably = new Ably.Rest('<YOUR-API-KEY>');
+```
+
+Next, let's go ahead and perform the actual REST request using the @request()@ method as follows:
+
+```[javascript]
+var URL = '/channels';
+var resultArea = document.getElementById('result');
+var channelCount = 0;
+//request a list of channels on button click
+function enumerateChannels() {
+    ably.request('get', URL, { limit: 100, direction: 'forwards' }, null, null, (err, results) => {
+        if (err) {
+            resultArea.value += 'An error occurred; err = ' + err.toString();
+        } else {
+            displayResults(results)
+            while (results.hasNext()) {
+                results.next(function (err, nextPage) {
+                    displayResults(nextPage);
+                });
+            }
```

Comment:
async functions don't work like that. This will be an infinite loop if hasNext is ever true. Not to mention your displayResults function does another http fetch to re-get the the first page of results again.

@Srushtika did you tag anyone as a reviewer for this pr? At least one other person should have reviewed this before it was merged

Reply:
@SimonWoolf , @tomczoink had a look at the tutorial, had suggested a few changes and I had updated the tutorial. However, there was no code review.

Reply:
For this tutorial in particular, did some fixes in https://github.com/ably/docs/pull/589.

I understand you did an `@channel` in slack asking for feedback, but I think that's maybe not an ideal workflow: things like this don't need literally everyone to all do a deep code review, but do need one of {me, paddy, matt} to do that. So there's a danger that if you just ask everyone in slack, everyone'll assume that someone else has done it. I think it's maybe better to pick one of us (maybe alternate between us) and tag them a reviewer in github (or maybe two for particularly important changes), and chase that person for a deep review. That seems to work well in other repos.
cc @mattheworiordan @paddybyers